### PR TITLE
Show and Restore archived playbooks

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
This PR allows users to see archived Playbooks in the backstage, and restore them.

##### TLDR:

**Server:**

- The playbook listing API now accepts a `with_archived` query (values: true/false; default false)
- The OpenAPI/Swagger file has been updated accordingly. 

**Webapp:**

- Adds a "With archived" filter
- Adds a restore option for the archived playbooks
- Show an icon+tooltip combo to indicate that a playbook is archived.

###### Screenshots:

<details><summary>Playbook list (new With archived filter + icon bellow archived playbooks)</summary>

image

</details>

<details><summary>Restore menu option on an archive playbook</summary>

image

</details>
<details><summary>Restore confirmation modal</summary>

image

</details>

***edit:*** The broken e2e test (`playbooks > overview > should copy playbook link`) seems to be present on master too

#### Ticket Link
N/A

#### Checklist

~~- [ ] Telemetry updated~~
~~- [ ] Gated by experimental feature flag~~
- [x] Unit tests updated
- [x] e2e tests added

### CODE REVIEW

**Review by crspeller — CHANGES_REQUESTED** [review]
Sorry for the long review cycle here. This kinda got dropped over the holidays. 

The PR looks good overall. It's going to need a bit of merging work given some of the changes. Feel free to reach out to me on community server if you need some help. 

Feel free to wait until Monday to get started on this. Welcome BTW 😄

**Comment by JulienTant** [comment]
> Sorry for the long review cycle here. This kinda got dropped over the holidays.

No problem!

> The PR looks good overall. It's going to need a bit of merging work given some of the changes. Feel free to reach out to me on community server if you need some help.
> Feel free to wait until Monday to get started on this. Welcome BTW smile

Thank you Christopher! I did it now as I don't quite know yet what's coming for me next week - you probably know I've not been assigned to the Playbooks team but to the Growth team.
I replicated what has been done on the "archive" part (modal on a dedicated `tsx` file).

***Note:*** Just confirmed with @cass.caplinger on the community server that the `archive` button does not refresh the page automatically. I remember hooking up on this to auto-refresh the list when a playbook was restored - so it's not working here anymore.

File: webapp/src/components/backstage/playbook_list_row.tsx [hunk 1]
```diff
@@ -70,38 +70,72 @@ const PlaybookItemRow = styled.div`
...
                     id={props.playbook.title}
                     text={props.playbook.title}
                 />
-                {props.displayTeam && <InfoLine>{teamName}</InfoLine>}
+                {infos.length > 0 &&
+                    <InfoLine>
+                        {infos.map((info, i) => (<Fragment key={props.playbook.id + '-infoline' + i}>
+                            {i > 0 && ' - '}
+                            {info}
+                        </Fragment>))}
```
Comment (crspeller) [inline webapp/src/components/backstage/playbook_list_row.tsx:115]: Nicer formatting? 
```suggestion
                        {infos.map((info, i) => (
                            <Fragment key={props.playbook.id + '-infoline' + i}>
                                {i > 0 && ' - '}
                                {info}
                            </Fragment>
                        ))}
```

**Review by crspeller — APPROVED** [review]
Known issue with the list not updating. Need to hook that up to websockets. So not a problem here. 

Thanks!

**Review by Phrynobatrachus — APPROVED** [review]
JS tests seem fine, it might be possible to drop some of the `data-testid`s in favor of accessibility options but I need to look into that more myself. 

Only notable thing for me:

- the modal for "archiving" still asks about "deleting"
- Screenshot from 2022-01-18 10-01-38
- might want change to "archive" for consistency? 
- **edit:** yes to changing the above per team meeting clarification this morning

**Comment by cpoile** [comment]
Hey @JulienTant, and welcome again! Sorry for the delay on reviewing this. Do you mind merging to master and addressing Cass's comment, then I'll take a look at it? (Let us know if you need help with the merge!)

**Comment by JulienTant** [comment]
> Hey @JulienTant, and welcome again! Sorry for the delay on reviewing this. Do you mind merging to master and addressing Cass's comment, then I'll take a look at it? (Let us know if you need help with the merge!)

Hi @cpoile, thanks! The wording has been changed archive and master has been merged into the branch.

**Review by Phrynobatrachus — APPROVED** [review]
thank you! sorry for the conflicting approval + textual 'request changes' from me before :face_with_spiral_eyes:

**Comment by cpoile** [comment]
@JulienTant Sorry again, but it looks like things keep changing underneath this PR. 😅

**Comment by JulienTant** [comment]
@cpoile Fixed 😅

File: server/api/playbooks.go [hunk 2]
```diff
@@ -470,7 +470,7 @@ func parseGetPlaybooksOptions(u *url.URL) (app.PlaybookFilterOptions, error) {
 
 	searchTerm := u.Query().Get("search_term")
 
-	withArchived := u.Query().Get("with_archived") == "true"
+	withArchived, _ := strconv.ParseBool(u.Query().Get("with_archived"))
```
Comment (JulienTant) [inline server/api/playbooks.go:473]: [ParseBool only returns an error when the input is not supported](https://github.com/golang/go/blob/2ebe77a2fda1ee9ff6fd9a3e08933ad1ebaea039/src/strconv/atob.go#L10), and returns false, which is acceptable in our use case.

**Review by cpoile — APPROVED** [review]
yay, thanks @JulienTant!
